datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -121,6 +123,27 @@ impl LogicalPlanBuilder {
         }))
     }
 
+    /// Convert a regular plan into a recursive query.
```

Comment:
Can we please document what `is_distinct` means here? I think it means "when updating the working table, should the results be deduplicated ('UNION') or not (UNION ALL)

Reply:
yup!

Reply:
added a doc comment and also beefed up the error message.